### ISSUE TITLE
 Few-shot setting in CoCoOp Experiments

### ISSUE THREAD

**Issue by Uzair-Khattak** [KaiyangZhou/CoOp#22]
Hello, 
Thank you for sharing your great work. 

I had a question regarding the few-shot setting in the CoCoOp experiments. In the paper, it is mentioned that CoCoOp follows a zero-shot evaluation (from base to novel classes) but for training the base classes, it uses a few-shot setting. However, generally for zero-shot evaluation, models are trained on the complete base classes. 

Does this mean that, CoCoOp and CoOp requires only a few-shot setting to perform well on novel categories. Can the same training recipe of CoCoOp or CoOp be used by training all examples of the base classes?

Thank you and kind regards.

**Comment by KaiyangZhou** [comment]
> Can the same training recipe of CoCoOp or CoOp be used by training all examples of the base classes

The key advantage of those methods is that they are data-efficient (i.e. with only a few shots the model can be adapted to a new task). Certainly you can use all data for tuning.

Data/parameter-efficient fine-tuning (or transfer learning) is more appealing than full-data fine-tuning (the latter has been commonly used) when it comes to adapting large vision foundation models like CLIP
